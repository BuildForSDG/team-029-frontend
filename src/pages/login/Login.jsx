import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { func, object } from 'prop-types';
import AppSpinner from '../../components/appSpinner/AppSpinner.jsx';
import Button from '../../components/button/Button.jsx';
import Input from '../../components/input/Input.jsx';
import Header from '../../components/header/Header.jsx';
import Alert from '../../components/alert/Alert.jsx';
import { loginUser as loginUserAction, clearError as clearErrorAction } from '../../redux/users/userAction';

import './Login.styles.scss';

const Login = ({ loginUser, userState, clearError, history }) => {
	const [formInput, setFormInput] = useState({
		email: '',
		password: ''
	})
	const { loginLoading, error } = userState;

	useEffect(() => {
		const { token } = userState;
		if (token) {
			return history.push("/dashboard");
		}
		return () => { }
	}, [userState])

	const onFormInputChange = (e) => {
		const { name, value } = e.target;
		return setFormInput({ ...formInput, [name]: value });
	}
	const onFormSubmit = (e) => {
		e.preventDefault();
		return loginUser(formInput);
	}

	return (
		<div className="login-page-wrapper">
			<form onSubmit={onFormSubmit} className="login-form">
				<Header title="Sign In" />
				{loginLoading && <AppSpinner />}
				{!loginLoading && error && (
					<Alert
						type="error"
						onClose={clearError}
						message="Invalid email or password"
						iconName="exclamation triangle"
					/>
				)}
				<Input
					onChange={onFormInputChange}
					inputName="email"
					label="Email or Staff ID"
					value={formInput.email}
					inputType="text"
					placeholder="your mail or staff ID"
				/>
				<Input
					label="Your password"
					inputName="password"
					inputType="password"
					placeholder="provide a password"
					onChange={onFormInputChange}
					value={formInput.password}
				/>
				<Button>Login</Button>
				<p className="reset-password">Reset password?</p>
			</form>
		</div>
	);
};

Login.propTypes = {
	loginUser: func.isRequired,
	userState: object.isRequired,
	clearError: func.isRequired,
	history: object.isRequired
}

const mapStateToProps = state => ({
	userState: state.user
});

const mapDispatchToProps = {
	loginUser: loginUserAction,
	clearError: clearErrorAction
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);
