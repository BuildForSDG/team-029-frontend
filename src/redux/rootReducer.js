import { combineReducers } from 'redux';
import { persistReducer } from 'redux-persist';

// selecting localstorage
import storage from 'redux-persist/lib/storage';

const persistConfig = {
  key: 'root',
  storage,
  whitelist: [] // what we want to persist
};

const rootReducer = combineReducers({});

export default persistReducer(persistConfig, rootReducer);
