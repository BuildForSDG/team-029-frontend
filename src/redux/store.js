import { persistStore } from 'redux-persist';
import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

import rootReducer from './rootReducer';

const middleWares = process.env.NODE_ENV === 'development' ? [logger, thunk] : [thunk];

export const store = createStore(rootReducer, applyMiddleware(...middleWares));
export const persistor = persistStore(store);
export default { persistor, store };
