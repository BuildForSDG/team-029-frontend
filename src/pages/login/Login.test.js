import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Login from './Login.jsx';
import { Provider } from 'react-redux';
import { store } from '../../redux/store'

afterEach(cleanup);

it('Should render <Login /> page component', () => {
	const mockedHistory = {
		push: () => { }
	}
	const { asFragment } = render(
		<Provider store={store}>
			<Login history={mockedHistory} />
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
