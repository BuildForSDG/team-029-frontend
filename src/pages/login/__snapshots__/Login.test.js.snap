// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <Login /> page component 1`] = `
<DocumentFragment>
  <div
    class="login-page-wrapper"
  >
    <form
      class="login-form"
    >
      <div
        class="heading-container undefined"
      >
        <h3
          class="undefined heading"
        >
          Sign In
        </h3>
        <div
          class="underline undefined"
        />
      </div>
      <div
        class="input-wrapper "
      >
        <label
          for="email"
        >
          Email or Staff ID
        </label>
        <input
          id="email"
          name="email"
          placeholder="your mail or staff ID"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="input-wrapper "
      >
        <label
          for="password"
        >
          Your password
        </label>
        <input
          id="password"
          name="password"
          placeholder="provide a password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        class="ui button  custom-button"
      >
        Login
      </button>
      <p
        class="reset-password"
      >
        Reset password?
      </p>
    </form>
  </div>
</DocumentFragment>
`;
